src/libraries/System.Data.Odbc/src/Common/System/Data/Common/DBConnectionString.cs

```diff
@@ -515,20 +515,20 @@ internal static string[] RemoveDuplicates(string[] restrictions)
                     count = 0;
                     for (int i = 0; i < restrictions.Length; ++i)
                     {
-                        if (null != restrictions[i])
+                        if (restrictions[i] is { } restriction)
```

Comment:
what does `is { }` mean?

Reply:
This is just a way to use pattern making to check if a value isn't null

Reply:
`!= null` or `is not null` would be better.

Reply:
`!= null` is the check being replaced, the problem is that since we're checking 

`is not null` is C# 9, right? Is it OK to use this already in runtime?

Reply:
FYI am going to merge as-is, but we can definitely replace these with the nicer `is not null` once we can use C# 9 here.

Reply:
What's the benefit of this complicated syntax over != null here?

Reply:
> is not null is C# 9, right? Is it OK to use this already in runtime

Yes and yes

Reply:
There's a tiny benefit of capturing `restriction`, that will be realized more when using `is not null`.

Reply:
Sure, but neither `{ }` nor `is not null` are necessary to achieve that:
```C#
if (restrictions[i] is string restriction)
{
    ...
}
```

Reply:
Ah, yeah that'd be better indeed.

Reply:
The slight disadvantage of `restrictions[i] is string restriction` is that it's ambiguous with regards to whether we're checking that something isn't null, or that it's a string (as opposed to something else). FWIW Resharper/Rider warn about this.

I tried `restrictions[i] is not null restriction`, assuming that's how the C# 9 feature looks like, but it doesn't seem to compile. I'll just switch to `restrictions[i] is string restriction` for now as suggested.

Reply:
If you see it as a disadvantage, then just do the super simple thing:
```C#
foreach (string? restriction in restrictions)
{
    if (restriction != null)
    {
        tmp[count++] = restriction;
    }
}
```
or
```C#
for (int i = 0; i < restrictions.Length; i++)
{
    string? restriction = restrictions[i];
    if (restriction != null)
    {
        tmp[count++] = restriction;
    }
}
```
or just suppress the warning:
```C#
for (int i = 0; i < restrictions.Length; i++)
{
    if (restrictions[i] != null)
    {
        tmp[count++] = restrictions[i]!; // TODO-NULLABLE: Indexer nullability tracked (https://github.com/dotnet/roslyn/issues/34644)
    }
}
```
as we do in dozens of other places in dotnet/runtime... the more we work around it, the less likely it'll be the compiler "fixes" it.
cc: @cston

Not everything needs to use fancy syntax.